### ISSUE TITLE
vip sudden change to external svc unavailable for parts serviceentry

### ISSUE THREAD

**Issue by zhj-hx** [istio/istio#44849]
### Bug Description

DNS-proxy is enabled
ISTIO_META_DNS_CAPTURE: "true"
ISTIO_META_DNS_AUTO_ALLOCATE: "true"

vip sudden change to external svc unavailable for parts serviceentry, ServiceEntry was not manually recreated during this time, completely changed by the system

serviceentry config
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: collect
spec:
  hosts:
    - collect.external
  ports:
    - number: 80
      name: http
      protocol: TCP
  exportTo:
    - "."
  resolution: STATIC
  location: MESH_EXTERNAL
  endpoints:
    - address: 10.111.110.27
      ports:
        http: 6070
```

 log：
[2023-05-10T07:07:44.397Z] "- - -" 0 - - - "-" 805 188 8 - "-" "-" "-" "-" "10.111.110.27:6070" outbound|80||collect.external
 172.27.96.218:51488 240.240.0.219:80 172.27.96.218:36856 - -

[2023-05-10T07:07:58.832Z] "GET /xxx HTTP/1.0" 502 - direct_response - "-" 0 0 0 - "1.1.1.1, 127.0.0.6" "Go-http-client/2.0" "d087d9bb-9aff-44ab-b589-27b2833d2faf" "collect.xxx.com" "-" - - 240.240.0.219:80 127.0.0.6:0 - block_all

DNS changes process
2023-05-10T07:07:44 
/ # ping collect.external
PING collect.external (240.240.0.219): 56 data bytes

2023-05-10T07:07:58
/ # ping collect.external
PING collect.external (240.240.0.218): 56 data bytes

similar questions，
serviceentry config
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: aaa-mysql
spec:
  hosts:
    - rm-xxx.mysql.rds.aliyuncs.com
  exportTo:
    - "."
  ports:
    - number: 3306
      name: tcp
      protocol: TCP
  resolution: DNS
  location: MESH_EXTERNAL
```

log:
[2023-05-10T07:07:59.034Z] "- - -" 0 - - - "-" 907 797 369 - "-" "-" "-" "-" "10.111.132.167:3306" outbound|3306||rm-xxx.mysql.rds.aliyuncs.com 172.27.102.143:48688 240.240.0.227:3306 172.27.102.143:42708 - -
[2023-05-10T07:08:04.560Z] "- - -" 0 UH - - "-" 0 0 0 - "-" "-" "-" "-" "-" BlackHoleCluster - 240.240.0.227:3306 172.27.102.143:429
60 - -
[2023-05-10T07:08:04.572Z] "- - -" 0 UH - - "-" 0 0 0 - "-" "-" "-" "-" "-" BlackHoleCluster - 240.240.0.227:3306 172.27.102.143:429
62 - -
[2023-05-10T07:08:11.366Z] "- - -" 0 - - - "-" 907 797 180 - "-" "-" "-" "-" "10.111.132.167:3306" outbound|3306||rm-xxx.mysql.rds.aliyuncs.com 172.27.102.143:49238 240.240.0.226:3306 172.27.102.143:50090 - -

DNS change process
Through the log, it can be found that the DNS resolution has changed

### Version

```Text
➜ istioctl version
client version: 1.13.2
control plane version: 1.10.0
data plane version: 1.10.0 (274 proxies)

➜ k version --short
Client Version: v1.24.0
Kustomize Version: v4.5.4
Server Version: v1.18.8-aliyun.1
```

### Affected product area

- [ ] Ambient
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [X] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

**Comment by howardjohn** [comment]
This has been fixed in 1.17. Istio 1.10 is long EOL

**Comment by zhj-hx** [comment]
> This has been fixed in 1.17. Istio 1.10 is long EOL

@howardjohn  Thanks , is the fix no longer maintained for older versions? Because we will not upgrade to version 1.17 in a short time

**Comment by zhj-hx** [comment]
> 

@howardjohn  Thanks, can you explain the specific reason for this bug?
